### DISCUSSION TITLE
Is there `asyncio` support for `get_data` or a recommend workaround?

### DISCUSSION THREAD

**Discussion by devraj** [betodealmeida/shillelagh#451] [Q&A]
I am wondering if `asyncio` is supported at the moment or if there are any plans to do so. At the moment I am just getting by using `asyncio.run` (rough implementation in the example), in the absence of `asyncio` are there any better examples of how to interface with `async` libraries?

I did see some issues/discussions around SQLAlchemy asyncio dialect support, I assume my discussion is along the same lines.

```py
    def get_data(  # pylint: disable=too-many-locals
        self,
        bounds: Dict[str, Filter],
        order: List[Tuple[str, RequestedOrder]],
        **kwargs: Any,
    ) -> Iterator[Row]:
        
        # TODO: refactor this to see if can support asyncio
        import asyncio
        cardholders = asyncio.run(Cardholder.list())
        # cardholders = await Cardholder.list()

        for row in cardholders.results:
            yield {
                "rowid": row.id,
                "id": row.id,
                "authorised": row.authorised,
                "first_name": row.first_name,
                "last_name": row.last_name,
            }
```

Appreciate you chiming in.

**Comment by betodealmeida** [comment]
There's been [some discussion](https://github.com/betodealmeida/shillelagh/issues/212) and it seems that this would be hard with SQLite. The Postgres backend ([multicorn](https://github.com/pgsql-io/multicorn2)) also doesn't seem to support it.
